BaikalDataGenerator/Source/render.cpp

```diff
@@ -0,0 +1,386 @@
+/**********************************************************************
+Copyright (c) 2018 Advanced Micro Devices, Inc. All rights reserved.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+********************************************************************/
+
+#include "utils.h"
+#include "render.h"
+#include "scene_io.h"
+#include "material_io.h"
+#include "SceneGraph/light.h"
+#include "Output/clwoutput.h"
+#include "BaikalIO/image_io.h"
+
+#include <filesystem>
+#include <fstream>
+#include "XML/tinyxml2.h"
+
+using namespace Baikal;
+
+namespace
+{
+    std::uint32_t constexpr kNumIterations = 4096;
+
+    bool operator != (RadeonRays::float3 left, RadeonRays::float3 right)
+    {
+        return (left.x != right.x) || (left.y != right.y) || (left.z != right.z);
+    }
+}
+
+struct OutputInfo
+{
+    Renderer::OutputType type;
+    std::string name;
+    // chanels number can be only 1 or 3
+    int chanels_num;
```

Comment:
chan**n**els_num. Also, we can rename to pure **channels**

Reply:
chenale_num is more clear name